flashinfer/comm/allreduce.py

```
@@ -223,9 +223,65 @@ def _mnnvl_workspace_check(
+            from cuda import cuda  # type: ignore[no-reuse-def]
+        from flashinfer.cuda_utils import checkCudaErrors
+
+        device_idx = torch.cuda.current_device()
+        multicast_supported = checkCudaErrors(
+            cuda.cuDeviceGetAttribute(
+                cuda.CUdevice_attribute.CU_DEVICE_ATTRIBUTE_MULTICAST_SUPPORTED,
+                device_idx,
+            )
+        )
+        return multicast_supported != 0
+    except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)